### PR TITLE
Stripe connect

### PR DESCRIPTION
**Pull request by MUGABA**
### What this PR does (required):
- This pr will help a merchant customer to create a stripe account and link

### Any information needed to test this feature (required):
- I User must have an active stripe account

### Any issues with the current functionality (optional):
- Implementation not test fully because I did not have an active stripe account.

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Hey @MUGABA, please link the ticket for this PR.

File: server/controllers/stripeConnect.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  });
+
+  await stripe.accountLinks.create({
+    account: id,
+    refresh_url: "http://localhost:3000/settings/payment-methods",
+    return_url: "http://localhost:3000/dashboard",
+    type: "account_onboarding",
+  });
+
+  return res.status(201).send({ success: petSitter });
```
Comment (ezeilo-su) [inline server/controllers/stripeConnect.js:55]: Please use a `try`/`catch` block

File: server/controllers/stripeConnect.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const PetSitter = require("../models/PetSitter");
+const stripe = require("stripe")(process.env.STRIPE_SECRET);
+
+// @route POST /connect/stripe
+// @desc user connects to stripe and creates account link
+// @access Public
+exports.stripeConnect = asyncHandler(async (req, res, next) => {
+  const { rate } = req.body;
+  const userId = req.user.id;
+  const profile = await Profile.findOne().where("userId").equals(userId);
```
Comment (ezeilo-su) [inline server/controllers/stripeConnect.js:15]: Please use a `try`/`catch` block

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Hey @MUGABA, good progress here. Please look at the comments I left and address them.

File: server/controllers/stripeConnect.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    capabilities: {
+      card_payments: { requested: true },
+      transfers: { requested: true },
+    },
+    company: {
+      name: profile.name,
+    },
+  });
+
+  const { id } = account;
```
[the remark this answers is not in the document]
Comment (MUGABA) [reply server/controllers/stripeConnect.js:39]: Thanks for the pointer I did test it and it works well

File: client/src/pages/Settings/Settings.tsx [hunk 4]
```diff
@@ -28,7 +29,7 @@ const settingsMenu = [
   {
     name: 'Payment methods',
     to: '/profile/settings/payment-methods',
-    component: <SettingHeader header="Payment Methods" />,
+    component: <StripeConnect />,
```
Comment (moffatethan) [inline client/src/pages/Settings/Settings.tsx:32]: This isn't for payment methods, let's just make a new settings page called 'Billing' or 'Stripe'

File: server/controllers/stripeConnect.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      email: currentUser.email,
+      default_currency: "CAD",
+      business_type: "individual",
+      company: {
+        name: currentUser.name,
+      },
+    });
+  } catch (err) {
+    res.status(400);
+    throw new Error("Stripe ccount not created", err);
```
Comment (moffatethan) [inline server/controllers/stripeConnect.js:33]:
```suggestion
    throw new Error("Stripe account not created", err);
```

File: server/controllers/stripeConnect.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  try {
+    await stripe.accountLinks.create({
+      account: account.id,
+      refresh_url: "http://localhost:3000/settings/payment-methods",
+      return_url: "http://localhost:3000/dashboard",
```
Comment (moffatethan) [inline server/controllers/stripeConnect.js:40]: Let's make these environment variables for easier production launch

File: server/controllers/stripeConnect.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    res.status(400);
+    throw new Error("Stripe account link did not create", err);
+  }
+
+  const rateInCents = rate * 100;
+  const petSitter = await PetSitter.create({
+    stripeConnectId: account.id,
+    rate: rateInCents,
+    userId,
+  });
```
Comment (moffatethan) [inline server/controllers/stripeConnect.js:53]: I think we should make a new page for editing the pet sitter fields

**Pull request merged**